python/revng/scripting/daemon_project.py

```diff
@@ -0,0 +1,216 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import json
+from typing import Any
+
+from gql import Client, gql
+from gql.transport.aiohttp import AIOHTTPTransport
+
+from revng.model import Binary, dump_diff_to_string, load_model_from_string, make_diff
+
+from .project import Project
+
+
+class DaemonProject(Project):
+    def __init__(self, url: str):
+        self.transport = AIOHTTPTransport(url=f"{url}/graphql/")
+        self.client = Client(
+            transport=self.transport, fetch_schema_from_transport=True, execute_timeout=None
+        )
+
+    def upload_binary(self, path: str):
+        query = gql(
+            """
+            mutation upload($file: Upload!) {
+                uploadFile(file: $file, container: "input")
+            }
+        """
+        )
+
+        with open(path, "rb") as binary_file:
+            result = self._make_request(query, {"file": binary_file}, upload_files=True)
+
+        if not result["uploadFile"]:
+            raise RuntimeError("File upload failed")
+
+        self.set_model(self._get_model())
+        self.parse_pipeline_description(self._get_pipeline_description())
```

Comment:
This should be done when first connecting to the server (e.g. `__init__`), not when uploading the binary